Прошу помочь исправить ошибки в работе

Lidia128 opened the pull request.

Lidia128 committed:
Прошу помочь исправить ошибки в работе

iStemmer commented on src/main/java/me/recipe/homework/service/impl/IngredientServiceImpl.java:
```diff
@@ -38,6 +43,20 @@ public Ingredient getIngredientById(int id) {
     }
+    @Override
+    public Path createIngredient(Ingredient ingredient){
+        Ingredient ingredient1 = ingredients.getOrDefault(ingredient, new Map<>());
```
тут логика потерялась, мы здесь должны просто создать новый ингредиент, зачем создавать Мапу? 

Этот метод про создание нового ингредиента или записи всех ингредиентов в файл?

iStemmer commented on src/main/java/me/recipe/homework/service/impl/RecipeServiceImpl.java:
```diff
@@ -45,6 +48,18 @@ public Recipe getRecipeById(int id) {
         }
     }
+    @Override
+    public Path createRicepe(Recipe recipe) throws IOException {
```
тут тоже самое. Это создание нового рецепта или сохранение?

Lidia128 committed:
Внесла небольшую корректировку кода и отформатировала все файлы, но в консоль выпадает ошбика, не могу понять что необходимо исправить, т.к. по файлу нигде ничего не подчеркнуто

Lidia128 committed:
Не могу понять почему в RecipeController в строке 23 подчёркнуто слово service (переменная создана, конструктор написан)

Lidia128 committed:
Удалила ключевое слово abstract в public abstract class RecipeServiceImpl implements RecipeService {,
но в RecipeController в строке 23 подчёркнуто слово service (переменная создана, конструктор написан).

Lidia128 committed:
Появляется ошика, но мне не понятно почему программа ищет поле в другом классе
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'recipeServiceImpl': Invocation of init method failed; nested exception is java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "quantityIngredient" (class me.recipe.homework.model.Recipe), not marked as ignorable (4 known properties: "ingredients", "name", "time", "steps"])

Lidia128 committed:
Начала править код, но появились новые ошибки в FileRecipeController. Запусаталсь окончательно

Lidia128 committed:
Начала править код, но появились новые ошибки в FileRecipeController.
ст.72 подчеркивает все красным
ст. 73 и 79 подчеркивает path
ст. 76 красная toFile

Lidia128 committed:
Внесла необходимые корректировки и всё заработало. Спасибо.

iStemmer commented on src/main/java/me/recipe/homework/service/impl/RecipeServiceImpl.java:
```diff
@@ -17,7 +17,7 @@
 
 @Service
-public class RecipeServiceImpl implements RecipeService {
+public abstract class RecipeServiceImpl implements RecipeService {
```
зачем абстракт?

iStemmer commented on src/main/java/me/recipe/homework/controller/FilesIngredientController.java:
```diff
@@ -14,7 +14,7 @@
 
 @RestController
-@RequestMapping("/files")
+@RequestMapping("/ingredient.files")
```
ingredient/files
или 
files/ingredients

iStemmer commented on src/main/java/me/recipe/homework/controller/FilesRecipeController.java:
```diff
@@ -12,12 +12,12 @@
 
 @RestController
-@RequestMapping("/files")
+@RequestMapping("/recipe.files")
```
recipe/files или files/recipe

Lidia128 committed:
Заменила . на / в контроллерах
